### ISSUE TITLE
Indentation: missing violations for '->' and ',' in return statement continuation

### ISSUE THREAD

**Issue by Tieshiyang** [checkstyle/checkstyle#17663]
I have read check documentation: https://checkstyle.sourceforge.io/checks/whitespace/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Code:
```bash
public class Test {
  int sum(int a, int b, int c) {
    return a
    + b     // Expected Violation (and Reported)
    + c;    // Expected Violation (and Reported)
  }

  java.util.function.Function<String, Integer> test_arrow() {
    return (String s) ->
    s.length(); // Expected Violation (but ignored)
  }

  int test_comma() {
    return sum(
    1, // Expected Violation (but ignored)
    2, // Expected Violation (but ignored)
    3  // Expected Violation (but ignored)
    );
  }
}
```
Config:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="localeLanguage" value="en"/>
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="2"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
    </module>
</module>
```
Cli:
```
$ java -jar checkstyle-10.26.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:4:5: 'method def' child has incorrect indentation level 4, expected level should be 8. [Indentation]
[ERROR] Test.java:5:5: 'method def' child has incorrect indentation level 4, expected level should be 8. [Indentation]
Audit done.
Checkstyle ends with 2 errors.
```

---

Observed behavior:
In return with +, indentation errors are correctly reported.
In return with lambda arrow -> or commas ,, no indentation violations are reported, even though they are also continuation lines.
This makes the behavior inconsistent: some continuation elements are checked, while others are ignored.

---

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/b79e1a54ecf29860f4c047a32acf77f0c5dfa7c8/src/main/resources/google_checks.xml#L308-L315

Sounds like you use Google style, if yes please share Google formatter output on provided code.
Example https://github.com/checkstyle/checkstyle/issues/17565

**Comment by VeniceWish** [comment]
Thanks for the reply.
From: https://google.github.io/styleguide/javaguide.html#s4.8.4-switch:~:text=A%20line%20is%20never,y)%3B%0A%20%20...%0A%7D

> A line is never broken adjacent to the arrow in a lambda or a switch rule, except that a break may come immediately after the arrow if the text following it consists of a single unbraced expression. 

Formatter:
```
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2025-08-24 16:28:05.721073800 +0800
+++ FormattedCode.java  2025-08-24 16:28:12.133093700 +0800
@@ -1,20 +1,19 @@
 public class Test {
   int sum(int a, int b, int c) {
     return a
-    + b     // Expected Violation (and Reported)
-    + c;    // Expected Violation (and Reported)
+        + b // Expected Violation (and Reported)
+        + c; // Expected Violation (and Reported)
   }

   java.util.function.Function<String, Integer> test_arrow() {
-    return (String s) ->
-    s.length(); // Expected Violation (but ignored)
+    return (String s) -> s.length(); // Expected Violation (but ignored)
   }

   int test_comma() {
     return sum(
-    1, // Expected Violation (but ignored)
-    2, // Expected Violation (but ignored)
-    3  // Expected Violation (but ignored)
-    );
+        1, // Expected Violation (but ignored)
+        2, // Expected Violation (but ignored)
+        3 // Expected Violation (but ignored)
+        );
   }
-}
\ No newline at end of file
+}
```

**Comment by aclfe** [comment]
@romani 

Is this issue open to be picked up? I noticed the previous PR has been inactive for 2 weeks. 
I want take this issue on

**Comment by aclfe** [comment]
I'll take this on, previous PR seems inactive. 
Thanks!

**Comment by nexpectArpit** [comment]
@romani @aclfe is this issue open for contribution?

**Comment by aclfe** [comment]
> [@romani](https://github.com/romani) [@aclfe](https://github.com/aclfe) is this issue open for contribution?

Implementation is finished, needs to send PR to external project. Main concern for external projects is the "why" factor, that's why they're not merging. Code following lambda operator isn't properly covered like other features of code, they question why only one line gets reduced in indentation and others aren't affected. It would be helpful for an indepth auditing of indentation for lambda functions to address the false negatives that we have

**Comment by romani** [comment]
We merged only tests extension for quick run and debug.
Everyone is welcome to continue this issue. Nobody is working on it now
